internal/services/mobilenetwork/mobile_network_data_network_resource.go

```diff
@@ -0,0 +1,241 @@
+			client := metadata.Client.MobileNetwork.DataNetworkClient
+
+			id, err := datanetwork.ParseDataNetworkID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model DataNetworkModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if resp.Model == nil {
+				return fmt.Errorf("retrieving %s: properties was nil", id)
+			}
+
+			properties := *resp.Model
+
+			if metadata.ResourceData.HasChange("description") {
+				if model.Description != "" {
+					properties.Properties.Description = &model.Description
+				} else {
+					properties.Properties.Description = nil
+				}
+			}
+
+			properties.SystemData = nil
+
+			if metadata.ResourceData.HasChange("tags") {
+				properties.Tags = &model.Tags
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, properties); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r DataNetworkResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MobileNetwork.DataNetworkClient
+
+			id, err := datanetwork.ParseDataNetworkID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if resp.Model == nil {
+				return fmt.Errorf("retrieving %s: model was nil", id)
+			}
+
+			model := *resp.Model
+
+			state := DataNetworkModel{
+				Name:                         id.DataNetworkName,
+				MobileNetworkMobileNetworkId: mobilenetwork.NewMobileNetworkID(id.SubscriptionId, id.ResourceGroupName, id.MobileNetworkName).ID(),
+				Location:                     location.Normalize(model.Location),
+			}
+
+			if properties := model.Properties; properties != nil {
+				if properties.Description != nil {
+					state.Description = *properties.Description
+				}
+			}
+			if model.Tags != nil {
+				state.Tags = *model.Tags
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r DataNetworkResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 180 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MobileNetwork.DataNetworkClient
+
+			id, err := datanetwork.ParseDataNetworkID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			if err := resourceMobileNetworkChildWaitForDeletion(ctx, id.ID(), func() (*http.Response, error) {
```

Comment:
is there an API bug tracking this, so we know when this is safe to remove?

It's worth noting that these workarounds shouldn't be needed in time with the new base layer, since that should both check the LRO and continue polling until the resource is fully provisioned/gone

Reply:
just opened an issue for this..https://github.com/Azure/azure-rest-api-specs/issues/22691

Reply:
Also, it might be better to add the workaround on delete function of `azurerm_mobile_network` instead of on children resources, WDYT?

Reply:
One of the design guarantees of Terraform is that it won't mark a resource as provisioned/gone until the resource is fully provisioned/removed - so we'd need this to be present within the child resources rather than the parent `azurerm_mobile_network` resource